core-metastore/src/main/java/org/zalando/nakadi/repository/db/EventTypeRepository.java

```diff
@@ -103,6 +121,25 @@ public EventType mapRow(final ResultSet rs, final int rowNum) throws SQLExceptio
                 new EventTypeMapper());
     }
 
+    public List<EventType> listEventTypes(final Set<String> eventTypeNames) {
+        final String query = String.format(
+                "SELECT et_event_type_object FROM zn_data.event_type WHERE zn_data.event_type.et_name IN ( %s )",
+                String.join(",", Collections.nCopies(eventTypeNames.size(), "?"))
+        );
+        return jdbcTemplate.query(
+                query,
+                eventTypeNames.toArray(),
+                new EventTypeMapper());
+    }
+
+    public <T> T lockingTable(final TableLockMode lockMode, final TransactionTemplate transactionTemplate,
+                              final TransactionCallback<T> action) {
+        return transactionTemplate.execute(ac -> {
+            jdbcTemplate.execute("LOCK TABLE zn_data.event_type IN " + lockMode.get() + " MODE");
+            return action.doInTransaction(ac);
```

Comment:
Why is this parameter called `ac`?  Should it be `status`?
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/support/TransactionCallback.html

Reply:
Thanks,  @a2exsh, for the intuitive suggestion, but
`ac` is quite fine here as it is intended to mean an action usually. See the documentation:
[https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/support/TransactionTemplate.html#execute-org.springframework.transaction.support.TransactionCallback-](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/support/TransactionTemplate.html#execute-org.springframework.transaction.support.TransactionCallback-)

It says right in the beginning: `Execute the action specified by the given callback object within a transaction. `
It again denotes:
`    action - the callback object that specifies the transactional action`

Reply:
@rishiraj88 
> `action - the callback object that specifies the transactional action`

Right.  But in our code this is what we pass to `execute`.  It implements `TransactionCallback` interface and `execute` calls our overridden `doInTransaction()` which takes a parameter `TransactionStatus status`.  That's why I think `ac` should be renamed to `status`.